tests/OmniSharp.Script.Tests/WorkspaceInformationTests.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Scripting.Hosting;
+using OmniSharp.Models.WorkspaceInformation;
+using TestUtility;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OmniSharp.Script.Tests
+{
+    public class WorkspaceInformationTests : AbstractTestFixture
+    {
+        private static Dictionary<string, string> s_netCoreScriptingConfiguration = new Dictionary<string, string>
+        {
+            ["script:enableScriptNuGetReferences"] = "true",
+            ["script:defaultTargetFramework"] = "netcoreapp2.1"
+        };
+
+        public WorkspaceInformationTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task SingleCsiScript()
+        {
+            using (var testProject = TestAssets.Instance.GetTestScript("SingleCsiScript"))
+            using (var host = CreateOmniSharpHost(testProject.Directory))
+            {
+                var workspaceInfo = await GetWorkspaceInfoAsync(host);
+                var project = Assert.Single(workspaceInfo.Projects);
+
+                Assert.Equal("main.csx", Path.GetFileName(project.Path));
+
+                // should have reference to mscorlib
+                VerifyCorLib(project);
+
+                // default globals object
+                Assert.Equal(typeof(CommandLineScriptGlobals), project.GlobalsType);
+            }
+        }
+
+        [Fact]
+        public async Task TwoCsiScripts()
+        {
+            using (var testProject = TestAssets.Instance.GetTestScript("TwoCsiScripts"))
+            using (var host = CreateOmniSharpHost(testProject.Directory))
+            {
+                var workspaceInfo = await GetWorkspaceInfoAsync(host);
+                var scriptProjects = workspaceInfo.Projects.ToArray();
+                Assert.Equal(2, scriptProjects.Length);
+
+                Assert.Equal("main.csx", Path.GetFileName(scriptProjects[0].Path));
```

Comment:
@filipw: I noticed that this test fails for me when run in the VS test runner. I believe this is because the order of the scripts in the project system is non-deterministic because they are held in a `ConcurrentDictionary`.

Reply:
yes - I noticed it this week too, sent a PR fixing another thing that additionally updates this test to ignore the order https://github.com/OmniSharp/omnisharp-roslyn/pull/1210